content/200-concepts/100-components/01-prisma-schema/06-relations/420-referential-integrity.mdx

```diff
@@ -8,21 +8,53 @@ preview: true
+Referential integrity is a property of a set of data that states that all its references are valid. It requires that if one record references another, then the referenced record must exist. For example, if a `Post` model defines an author, then the author must also exist. Referential integrity is enforced through constraints that prevent you making changes that break references, and through defining [referential actions](/concepts/components/prisma-schema/relations/referential-actions) that run when updating or deleting records. In the case of the `Post` model, for example, you will need to define what happens if its author is deleted.
 
-The referential integrity can either be handled in the database using foreign keys, or in the Prisma Client. There are pros and cons on both. In general, if the database supports foreign keys, it is usually the preferred choice and the default for SQL databases when using Prisma.
+In Prisma, referential integrity is implemented by defining relations between records using the [`@relation`](/reference/api-reference/prisma-schema-reference#relation) attribute. The `onUpdate` and `onDelete` arguments of this attribute can be used to define which referential action take place on updating or deleting the related record.
 
-In certain cases, either when the database doesn't support foreign keys, or when they are not preferred due to scaling reasons, Prisma can disable them and handle the referential integrity in the client.
+For example, take the following blog post schema, with a one-to-many relation between `User`s and `Post`s:
+
+```prisma file=schema.prisma highlight=4;normal
+model Post {
+  id       Int    @id @default(autoincrement())
+  title    String
+  author   User   @relation(fields: [authorId], references: [id], onDelete: Cascade)
+  authorId Int
+}
+
+model User {
+  id    Int    @id @default(autoincrement())
+  posts Post[]
+}
+```
+
+In this case, the `onDelete: Cascade` argument specifies the `Cascade` referential action, deleting the post if the related user is deleted. See [Referential actions](/concepts/components/prisma-schema/relations/referential-actions) for further details of supported actions.
+
+When connecting to relational databases that support foreign keys, referential integrity can be handled in the underlying database. This is the default option in Prisma when using SQL databases.
+
+Alternatively, Prisma can emulate referential integrity in the client. This is the default option when using MongoDB as a database, because it does not support foreign keys. It can also be selected for relational databases in cases where foreign keys are not supported, or where it preferred for other reasons (for example, in cases where there are scaling concerns).
+
+<Admonition type="warning">
+  Be aware that currently Prisma only implements the referential actions when
+  emulating referential integrity in the client, and not the constraints. If you
```

Comment:
We have no definition of what "the contraints" are before it is mentioned here (Should probably be something along the lines of "Foreign keys also create constraints, which make it impossible to manipulate data in a way that these constraints would be violated. Instead of executing the query, the database responds with an error. An example: A foreign key column always has to have an actually existing value from the target field")

Reply:
See comment here: https://github.com/prisma/docs/pull/2581#discussion_r771260070

Reply:
Link unfortunately does not go anywhere for me :(